src/InstagramClone.Api/Database/InstagramCloneDbContext.cs

```diff
@@ -0,0 +1,37 @@
+﻿using InstagramClone.Api.Entities;
+using Microsoft.EntityFrameworkCore;
+
+namespace InstagramClone.Api.Database
+{
+    public class InstagramCloneDbContext : DbContext
+    {
+        public InstagramCloneDbContext(): base()
+        {
+
+        }
+
+       
+
+        protected override void OnModelCreating(ModelBuilder modelBuilder)
+        {
+            modelBuilder.Entity<User>().HasKey(u => u.Id);
+            modelBuilder.Entity<Post>().HasKey(p => p.Id);
+            modelBuilder.Entity<PostReaction>().HasKey(pr => pr.Id);
+            modelBuilder.Entity<UserFollower>().HasKey(userFollower => userFollower.FollowedUserId);
+
+
+
+            modelBuilder.Entity<Post>().HasOne<User>(post => post.Author).WithMany( user => user.Posts);
```

Comment:
You should do this same thing for all the 1 to many relationships among your entities.

i.e. first you should 

1. identify all the other 1 to many relationships
2. does the entity (the 1) have the many navigation property? (i.e. the Icollection navigation property of the many). if it does not, then add it
3. after adding the Icollection navigation property for the "many" relationship, then comeback here to configure the relationship using `HasOne` + `WithMany`

Reply:
fixed

Reply:
one more thing, for each of the one to many, you should tell entity framework about the foreignkey property that we have configure. See this article for how to do it.

https://www.entityframeworktutorial.net/efcore/configure-one-to-many-relationship-using-fluent-api-in-ef-core.aspx